torch_geometric/transforms/add_metapaths.py

```diff
@@ -192,3 +193,187 @@ def _get_edge_weight(self, data: HeteroData,
+            types will be dropped. (default: :obj:`False`)
+        keep_same_node_type (bool, optional): If set to :obj:`True`, existing
+            edge types between the same node type are not dropped even in case
+            :obj:`drop_orig_edges` is set to :obj:`True`.
+            (default: :obj:`False`)
+        drop_unconnected_nodes (bool, optional): If set to :obj:`True` drop
+            node types not connected by any edge type. (default: :obj:`False`)
+        walks_per_node (int, List[int], optional): The number of walks to
+            sample for each starting node in a metapth. (default: :obj:`1`)
+        sample_ratio (float, optional): The ratio of source nodes to start
+            random walks from. (default: :obj:`1.0`)
+    """
+    def __init__(
+        self,
+        metapaths: List[List[EdgeType]],
+        drop_orig_edges: bool = False,
+        keep_same_node_type: bool = False,
+        drop_unconnected_nodes: bool = False,
+        walks_per_node: Union[int, List[int]] = 1,
+        sample_ratio: float = 1.0,
+    ):
+
+        for path in metapaths:
+            assert len(path) >= 2, f"Invalid metapath '{path}'"
+            assert all([
+                j[-1] == path[i + 1][0] for i, j in enumerate(path[:-1])
+            ]), f"Invalid sequence of node types in '{path}'"
+
+        self.metapaths = metapaths
+        self.drop_orig_edges = drop_orig_edges
+        self.keep_same_node_type = keep_same_node_type
+        self.drop_unconnected_nodes = drop_unconnected_nodes
+        self.sample_ratio = sample_ratio
+        if isinstance(walks_per_node, int):
+            walks_per_node = [walks_per_node] * len(metapaths)
+        else:
+            assert len(walks_per_node) == len(metapaths)
+        self.walks_per_node = walks_per_node
+
+    def __call__(self, data: HeteroData) -> HeteroData:
+        edge_types = data.edge_types  # save original edge types
+        data.metapath_dict = {}
+
+        for j, metapath in enumerate(self.metapaths):
+            for edge_type in metapath:
+                assert data._to_canonical(
+                    edge_type) in edge_types, f"'{edge_type}' not present"
+
+            src_node = metapath[0][0]
+            num_nodes = data[src_node].num_nodes
+            num_starts = round(num_nodes * self.sample_ratio)
+            row = start = torch.randperm(num_nodes)[:num_starts].repeat(
+                self.walks_per_node[j])
+
+            for i, edge_type in enumerate(metapath):
+                edge_index = data[edge_type].edge_index
+                adj = SparseTensor(row=edge_index[0], col=edge_index[1],
+                                   sparse_sizes=data[edge_type].size())
+                col, mask = self.sample(adj, start)
+                row, col = row[mask], col[mask]
+                start = col
+
+            new_edge_type = (metapath[0][0], f'metapath_{j}', metapath[-1][-1])
+            data[new_edge_type].edge_index = coalesce(torch.vstack([row, col]))
```

Comment:
Set `edge_weight` as `1`, this will count the number of `metapaths` between two nodes and set that as the `edge_weight` if `weighted` is `True`. I am guessing you were going to do this anyway.

Reply:
Exactly. Yet, the problem is, `edge_weight` is not really the number of metapaths between two nodes and its value depends on `walks_per_node`. This would potentially bring some inconsistency between `AddMetapaths` and `AddRandomMetaPaths`.

Reply:
In `AddMetapaths` the `edge_weight` is either the sum of weights of all metapaths between two nodes or the number of metapaths between two nodes. So setting in to number of random metapaths isn't inconsistent right?

Reply:
Yep. Maybe we should drop the argument `weighted` in `AddRandomMetaPaths`? WDYT?

Reply:
I am ok with not supporting it in this PR, we could add a todo comment to add `weighted` and address it in another PR.

Reply:
Thanks for your feedback @wsad1. I think I have nothing else to do for this PR. Could you please check it again at your convenience @rusty1s ?